iothub/service/src/Messaging/FileUploadNotificationProcessorClient.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Sockets;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Devices.Common;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Common.Extensions;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Subclient of <see cref="IotHubServiceClient"/> for handling file upload notications.
+    /// </summary>
+    /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-file-upload#service-file-upload-notifications"/>.
+    public class FileUploadNotificationProcessorClient : IDisposable
+    {
+        private readonly string _hostName;
+        private readonly IotHubConnectionProperties _credentialProvider;
+        private readonly IotHubConnection _connection;
+
+        /// <summary>
+        /// The callback to be executed each time file upload notification is received from the service.
+        /// </summary>
+        /// <remarks>
+        /// May not be null.
+        /// </remarks>
+        public Func<FileNotification, DeliveryAcknowledgement> _fileNotificationProcessor;
+
+        /// <summary>
+        /// The callback to be executed when the connection is lost.
+        /// </summary>
+        public Action<ErrorContext> _errorProcessor;
```

Comment:
Public facing properties and fields should be CamelCase.

We only name private fields with an underscore and camelCase.

```suggestion
        public Action<ErrorContext> ErrorProcessor;
```